myhoard/controller.py

```diff
@@ -669,7 +669,12 @@ def _binlog_uploaded(self, *, local_index, remote_key, stream):
 
     def _build_backup_stream(self, backup):
         stream_id = backup["stream_id"]
-        backup_site = self.backup_sites[backup["site"]]
+        backup_site_name = backup["site"]
+        try:
+            backup_site = self.backup_sites[backup_site_name]
+        except KeyError:
+            self.stats.increase("myhoard.unknown_backup_site", tags={"backup_site": backup_site_name})
+            raise AssertionError(f"backup site {backup_site_name} unknown (sites: {list(self.backup_sites.keys())!r})")
```

Comment:
I would change this to a specific exception so if this ends up in a log it's clear from the exception what could happen. Helps with classifying errors without stats.

Reply:
Done.